locksmith/src/models/sequelize.ts

```diff
@@ -0,0 +1,53 @@
+/* eslint-disable no-redeclare */
+
+import {
+  ModelAttributeColumnOptions,
+  DataType as SequelizeDataType,
+  ModelOptions,
+  ModelValidateOptions,
+  Model,
+  DataTypes,
+  BelongsToOptions,
+} from 'sequelize'
+
+interface TableOptions extends ModelOptions {
+  modelName?: string
+  /**
+   * Enable optimistic locking.  When enabled, sequelize will add a version count attribute
+   * to the model and throw an OptimisticLockingError error when stale instances are saved.
+   * Set to true or a string with the attribute name you want to use to enable.
+   */
+  version?: boolean | string
+}
+
+export declare function Table(options: TableOptions): Function
+export declare function Table(target: Function): void
+
+export declare function Column(dataType: SequelizeDataType): Function
+export declare function Column(
+  options: Partial<ModelAttributeColumnOptions>
+): Function
+export declare function Column(
+  target: any,
+  propertyName: string,
+  propertyDescriptor?: PropertyDescriptor
+): void
+
+export declare function Validate(options: ModelValidateOptions): Function
+
+export declare function AllowNull(target: any, propertyName: string): void
+export declare function AllowNull(allowNull: boolean): Function
+
+type ModelClassGetter = (returns?: void) => typeof Model
+export declare function BelongsTo(
+  associatedClassGetter: ModelClassGetter,
+  foreignKey?: string
+): Function
+export declare function BelongsTo(
+  associatedClassGetter: ModelClassGetter,
+  options?: BelongsToOptions
+): Function
+
+export declare const DataType: typeof DataTypes
```

Comment:
Could you explain what you are doing do here eventually?

Are you creating new decorators out of sequelize?

Reply:
yes, because `sequilize` don't provide all model that we currently get from `sequilize-typescript`, i basically get the type definition provided by `sequilize-typescript` and defined it in this file.

eg.
for `Model` and `Column` are not exported from `sequilize`

Reply:
yeah, because there is a different way to use typescript with sequelize. Decorators are part of the sequelize-typescript package so declaring types for them is not adding anything. You should remove them completely. 

Read this typescript page on sequelize: https://sequelize.org/master/manual/typescript.html

cc @julien51